.github/workflows/typing.yml

```diff
@@ -0,0 +1,32 @@
+# This workflow is running types checks using `tox -e typing` you could do locally.
+# As type hinting is not yet fully and consistently done through out the project only
+# some (randomly) selected files (listed in tox.ini) will be checked.  Feel welcome
+# to add extra files to the list there. See https://github.com/datalad/datalad/issues/6884
+# for the overall goal/progress towards type hints coverage in DataLad.
+name: Type-check
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  typing:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade tox
+
+      - name: Run type checker
+        run: tox -e typing
```

Comment:
FWIW -- we better adopt workflow like https://github.com/dandi/zarr_checksum/blob/2a30d1b482e96aa475f0a480f813be6ab8f9a1c9/.github/workflows/test.yml -- I like how it centralizes other testing as well to that single workflow + tox. WDYT?